galerautils/src/gu_threads.c

```diff
@@ -248,11 +248,11 @@ int gu_barrier_init_SYS (gu_barrier_t_SYS *barrier,
         errno = EINVAL;
         return -1;
     }
-    if(gu_mutex_init_SYS (&barrier->mutex, 0) < 0)
```

Comment:
there was a patch 09848b6f7 by Jan from 2021-11-11 14:22:24 +0200 that swapped the order of the arguments to `gu_mutex_init_SYS`. so now its `gu_mutex_init_SYS(const wsrep_mutex_key_t* key, gu_mutex_t_SYS *mutex)` so that for keyless mutexes its using `(NULL, &mutex)`.

the macros, if you mean `gu_mutex_init` is just a debug wrapper over the name, the only that I know: `#define gu_mutex_init            gu_mutex_init_SYS`.

Reply:
the patch replaced `- #define gu_mutex_init_SYS     pthread_mutex_init` with its own impl that has different args (key, mutex)

Reply:
https://github.com/MariaDB/galera/pull/25